### ISSUE TITLE
Dataset and Trimap

### ISSUE THREAD

**Issue by carfei** [ZHKKKe/MODNet#29]
Hello! I used the model you trained to test a large number of pictures, and found that there is a significant problem. When there is an intermediate background such as a seat and other people’s avatars at both ends of the head and shoulders, the matting effect is poor, I can understand Is the semantic segmentation of portraits in the network inaccurate? Or there is no such case in your improved portrait dataset, which leads to poor segmentation. If this is the problem, can you add some such images to it?
You mentioned that the modnet can use trimap input, how do you do it? If the trimap is not very accurate, can the network adapt to the trimap to some extent?
There is also a portrait semantic segmentation model in the model you provided. How was it trained? It was in the first stage of the network.
![11](https://user-images.githubusercontent.com/50562084/102499017-a667a280-40b5-11eb-8cc1-de140044c1d2.png)
![11-1](https://user-images.githubusercontent.com/50562084/102499031-a9fb2980-40b5-11eb-8623-abf9a132049e.png)
![22](https://user-images.githubusercontent.com/50562084/102499115-c1d2ad80-40b5-11eb-8506-093c4a39ed96.png)
![22-1](https://user-images.githubusercontent.com/50562084/102499121-c39c7100-40b5-11eb-9a5d-9d621fc99840.png)
![33](https://user-images.githubusercontent.com/50562084/102499132-c5fecb00-40b5-11eb-87e2-6b4cc4430735.png)
![33-1](https://user-images.githubusercontent.com/50562084/102499140-c8612500-40b5-11eb-9053-516cac04aee9.png)
![9](https://user-images.githubusercontent.com/50562084/102499156-cbf4ac00-40b5-11eb-9503-7b5a898443c2.png)
![11](https://user-images.githubusercontent.com/50562084/102499171-d3b45080-40b5-11eb-8d63-29b794e9ce3d.png)
![3](https://user-images.githubusercontent.com/50562084/102499187-d7e06e00-40b5-11eb-80ab-1d550776e60d.png)
![3](https://user-images.githubusercontent.com/50562084/102499211-ddd64f00-40b5-11eb-9962-193c1f33d8b0.png)

**Comment by ZHKKKe** [comment]
Hi, thanks for your attention.

For your questions:

**Q1: When there is an intermediate background such as a seat and other people’s avatars at both ends of the head and shoulders, the matting effect is poor, I can understand Is the semantic segmentation of portraits in the network inaccurate?**
Yes. There are two main problems to cause inaccurate mattes. **First**, these cases are not included in our training set, so the portrait semantic segmentation is wrong. **Second**, our pre-trained model for image matting is not adapted to the domain of daily photos. The pre-trained models used in our WebCam demo may obtain better performance in your cases (your images appear to contain noise). To achieve this, in our [Colab image matting demo](https://colab.research.google.com/drive/1GANpbKT06aEFiW-Ssx0DQnnEADcXwQG6?usp=sharing), please change:
```
!gdown --id 1mcr7ALciuAsHCpLnrtG_eop5-EYhbCmz \
```
to 
```
!gdown --id 1Nf1ZxeJZJL8Qx9KadcYYyEmmlKhTADxX \
```

**Q2: If this is the problem, can you add some such images to it?**
I am sorry that we do not have the labeled data of such images. One possible solution is to perform SOC adaptation by using the unlabeled data. However, SOC can not address the problems caused by inaccurate semantic segmentation.

**Q3: You mentioned that the modnet can use trimap input, how do you do it? If the trimap is not very accurate, can the network adapt to the trimap to some extent?**
If you want to use the trimap, you need to concatenate it with the RGB image as the input, which means the model should be re-trained. If you have trimap, I suggest you try the existing trimap-based methods, e.g., [DIM](https://github.com/foamliu/Deep-Image-Matting-PyTorch).

**Q4: There is also a portrait semantic segmentation model in the model you provided. How was it trained?**
This model is the pre-trained MobileNetV2 from [this repo](https://github.com/ZHKKKe/Semantic_Human_Matting). We initial our backbone by its weights for training.
